### DISCUSSION TITLE
[axios.defaults.withCredentials = true] equivalent in Nuxt 3?

### DISCUSSION THREAD

**Discussion by Arturexe** [nuxt/nuxt#16171] [Questions]
I'm trying to send an authenticate client request to a `laravel backend` (laravel sanctum) but am getting CORS error:
```
Access to fetch at 'http://127.0.0.1:8000/api/authenticate' from origin 'http://localhost:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

To resolve the same issue in `vue 3`, axios can be configured like so:

axios.defaults.withCredentials = true

but how would I do it in `Nuxt 3`?

I've been using the `$fetch` API to send the auth request:
```
$fetch('http://127.0.0.1:8000/api/authenticate', {
        method: 'POST',
        body: form
    }).then((res) => {console.log(res)})
```

**Comment by TheAlexLichter** [comment]
You can set `credentials:"include"` in the options.

```js
$fetch('http://example.com', {
  credentials: 'include'
})
```

**Reply by Arturexe** [reply]
Hey, when I use `credentials: 'include'` this error message is shown (something with wildcard at the end):
```
Access to fetch at 'http://127.0.0.1:8000/api/authenticate' from origin 'http://localhost:3000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'.
```

Has it something to do with `config/cors.php` ?
```
<?php

return [

    'paths' => ['api/*', 'sanctum/csrf-cookie'],

    'allowed_methods' => ['*'],

    'allowed_origins' => ['*'],

    'allowed_origins_patterns' => [],

    'allowed_headers' => ['*'],

    'exposed_headers' => [],

    'max_age' => 0,

    'supports_credentials' => false,

];
```

**Reply by TheAlexLichter** [reply]
That is indeed a requirement when using above mentioned options. Check https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors/CORSNotSupportingCredentials

**Reply by Arturexe** [reply]
So... I've been playing with the [laravel cors package](https://www.positronx.io/how-to-enable-cors-in-laravel/) for some time now and I don't know why it's not working. Same error message as in original question.
I don't understand why it works with axios by setting one prop to true but in nuxt 3 there seem to be much more work involved.

**Reply by TheAlexLichter** [reply]
Do you have an example API endpoint I could try out?
